### PR TITLE
2263-Create category endpoint and unit tests

### PR DESCRIPTION
**Pull request by Akrish17**
2263-Create category endpoint and unit tests

📲 What
Endpoint created to add a category to menu

🤔 Why
Need to be able to create a new Category

🛠 How
Add a category to an existing menu

👀 Evidence

🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java [hunk 1]
```diff
@@ -74,7 +80,7 @@ public MenuControllerImpl(MenuService menuService, ObjectMapper mapper) {
     ResourceCreatedResponseDto createdResponse =
```
Comment (peet-j) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java:80]: Add a constructor to ResourceCreatedResponseDto

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/dto/requestDto/CreateCategoryRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ */
+
+@Schema(name = "CreateCategoryRequest")
+public class CreateCategoryRequestDto {
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  @JsonPropertyOrder({
+          "name",
+          "description"
+  })
```
Comment (peet-j) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/dto/requestDto/CreateCategoryRequestDto.java:24]: Is the above needed when there's already validation on the attributes themselves?

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java [hunk 3]
```diff
@@ -90,4 +96,40 @@ private Menu convertMenuDtoToMenu(MenuCreateRequestDto requestDto) {
...
+   * @param requestDto dto mapping Create category request
+   * @return CategoryCreateResponse
+   */
+  @Override
+  public ResponseEntity<ResourceCreatedResponseDto>
+  createCategory(UUID id, CreateCategoryRequestDto requestDto) {
+    Category category = convertCategoryDtoToCategory(requestDto);
+
+    if (isNull(category)) {
+      return new ResponseEntity<>(BAD_REQUEST);
```
Comment (peet-j) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java:110]: Is this going to happen? The spring validation will catch any data issues no?

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java [hunk 4]
```diff
@@ -90,4 +96,40 @@ private Menu convertMenuDtoToMenu(MenuCreateRequestDto requestDto) {
...
+    }
+    category = this.menuService.saveCategory(id, category);
+
+    if (isNull(category)) {
+      return new ResponseEntity<>(NOT_FOUND);
```
Comment (peet-j) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java:115]: Implict "menu not found"?

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java [hunk 5]
```diff
@@ -90,4 +96,40 @@ private Menu convertMenuDtoToMenu(MenuCreateRequestDto requestDto) {
...
+    }
+
+    ResourceCreatedResponseDto createdResponse =
+            new ResourceCreatedResponseDto();
+    createdResponse.setId(category.getId());
```
Comment (peet-j) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java:120]: As above, a constructor would be far neater

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java [hunk 6]
```diff
@@ -90,4 +96,40 @@ private Menu convertMenuDtoToMenu(MenuCreateRequestDto requestDto) {
...
+
+  /**
+   * Mapping MenuRequestDto to Menu object
+   *
+   * @param requestDto Menu request dto
+   * @return mapped menu
+   */
+  private Category convertCategoryDtoToCategory(
+          CreateCategoryRequestDto requestDto) {
+    return this.mapper.convertValue(requestDto, Category.class);
```
Comment (peet-j) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/api/v1/menu/impl/MenuControllerImpl.java:133]: `ObjectMapper` has been superseded by `JsonMapper` in the latest jackson

File: java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/service/CosmosHelper.java [hunk 7]
```diff
@@ -5,7 +5,10 @@
 
 public class CosmosHelper {
 
-    public static CosmosPageRequest pageRequestWithSort(Sort.Direction direction, String variable, int pageNo, int pageSize) {
+    private CosmosHelper() {}
```
Comment (peet-j) [inline java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/service/CosmosHelper.java:8]: have deleted this on my branch. seems unnecessary

**Comment by Akrish17** [comment]
Closing this since these changes are now merged into master from #32

**Pull request closed without merging**